pkgs/tools/misc/shelldap/default.nix

```diff
@@ -0,0 +1,24 @@
+{ stdenv, fetchurl, perlPackages }:
+perlPackages.buildPerlPackage rec {
+  name = "shelldap-${version}";
+  version = "1.4.0";
+  src = fetchurl {
+    url = "https://bitbucket.org/mahlon/shelldap/downloads/shelldap-${version}.tar.gz";
+    sha256 = "07gkvvxcgw3pgkfy8p9mmidakciaq1rsq5zhmdqd8zcwgqkrr24i";
+  };
+  buildInputs = with perlPackages; [ perl YAMLSyck NetLDAP AlgorithmDiff IOSocketSSL AuthenSASL TermReadLineGnu TermShell ];
+  prePatch = ''
+    touch Makefile.PL
+  '';
+  installPhase = ''
+    mkdir -p $out/bin
+    cp shelldap $out/bin
```

Comment:
1. You can use ```install``` instead of explicitly calling mkdir and cp.
2. Please also run the ```preInstall``` and ```postInstall``` hooks

Reply:
Ok thank you. I updated the code.